federated_content_connector/filters/pipeline.py

```diff
@@ -76,3 +79,46 @@ def run_filter(self, course_key, serialized_enrollment):  # pylint: disable=argu
             pass
 
         return {'course_key': course_key, 'serialized_enrollment': serialized_enrollment}
+
+
+class CreateApiRenderCourseRunStep(PipelineStep):
+    """
+    Step that modifies the courserun data for the course.
+
+    Example usage:
+
+    Add the following configurations to your configuration file:
+
+        "OPEN_EDX_FILTERS_CONFIG": {
+            "org.openedx.learning.home.courserun.api.rendered.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "federated_content_connector.filters.pipeline.CreateApiRenderCourseRunStep"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, serialized_courserun):  # pylint: disable=arguments-differ
+        """
+        Pipeline step that modifies the courserun data for the course.
+        """
+        try:
+            course_details = CourseDetails.objects.get(id=serialized_courserun.get('courseId'))
+            course_type = course_details.course_type
+            product_source = course_details.product_source
+            start_date, end_date = course_details.start_date, course_details.end_date
+
+            if product_source == PRODUCT_SOURCE_2U and course_type == EXEC_ED_COURSE_TYPE:
+                now_utc = datetime.now(utc)
+                serialized_courserun.update({
```

Comment:
one more thing. Should we make a deep copy of `serialized_courserun` and then update it?

Reply:
The serialized_courserun dictionary contains basic string and boolean values, and there are no nested objects or complex structures. Making a deep copy here would not provide any significant benefit, as the dictionary itself is simple and directly updatable. So that is why I choose shallow copy.